src/dotnet/Chat.ML/AuthorNameRetriever.cs

```diff
@@ -0,0 +1,24 @@
+namespace ActualChat.Chat.ML;
+
+internal interface IAuthorNameRetriever
+{
+    Task<string> GetAuthorName(AuthorId authorId);
+}
+
+internal sealed class DefaultAuthorNameRetriever(IAuthorsBackend authorsBackend) : IAuthorNameRetriever
+{
+    public async Task<string> GetAuthorName(AuthorId authorId)
+    {
+        var author = await authorsBackend
+            .Get(authorId.ChatId, authorId, AuthorsBackend_GetAuthorOption.Full, default)
+            .ConfigureAwait(false);
+        var authorName = author?.Avatar.Name ?? "author-" + authorId.LocalId;
+        return authorName;
+    }
+}
+
+internal sealed class DelegateAuthorNameRetriever(Func<AuthorId, Task<string>> del) : IAuthorNameRetriever
+{
+    public  Task<string> GetAuthorName(AuthorId authorId)
+        => del(authorId);
```

Comment:
del sounds dangerous))
maybe change del to underlying, nameRetriever, nameGetter or something else

Reply:
changed